src/lib.rs

```diff
@@ -1253,7 +1253,12 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 			PaymentParameters::from_node_id(node_id, self.config.default_cltv_expiry_delta),
 			amount_msat,
 		);
-		let recipient_fields = RecipientOnionFields::spontaneous_empty();
+		let recipient_fields = RecipientOnionFields::spontaneous_empty()
+			.with_custom_tlvs(custom_tlvs.into_iter().map(|tlv| (tlv.r#type, tlv.value)).collect())
+			.map_err(|_| {
+				log_error!(self.logger, "Payment error: invalid custom TLVs.");
+				Error::PaymentSendingFailed
```

Comment:
Should we add `Error::InvalidCustomTLVs` and should the error be logged instead of `|_|`?

Reply:
There's nothing to log, unfortunately, since `with_custom_tlvs()` returns an empty type for error. I agree we could add a new enum value for that error, though